server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java

```diff
@@ -44,27 +56,82 @@ public void define(NewController context) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    JsonWriter json = response.newJsonWriter().beginObject();
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      Optional<UserDto> user = selectCurrentUser(dbSession);
```

Comment:
instead of checking presence of login in both `selectCurrentUser()` and `selectGroups()`, I suggest:
```
if (userSession.isLoggedIn()) {
  // select user and groups
}
```